Falcon/Falcon/Core/AssetManager.cpp

```diff
@@ -50,12 +55,119 @@ Mesh* AssetManager::GetMesh( const std::string& path)
 	return mesh;
 }
 
+Mesh* AssetManager::LoadTerrain(const std::string& path)
+{
+	//Get file data
+	char* json = nullptr;
+	int32_t size;
+	std::ifstream jsonFile(path, std::ios::in | std::ios::ate | std::ios::binary);
+	if (jsonFile.is_open()) {
+		size = jsonFile.tellg();
+		jsonFile.seekg(std::ios::beg);
+		json = fmemory::fnew_arr<char>(size + 1);
+		jsonFile.read(json, size);
+		json[size] = 0;
+		jsonFile.close();
+	}
+
+	//Start json doc
+	rapidjson::Document doc;
+	doc.Parse(json);
+	fmemory::fdelete<char>(json);
+
+
+	if (doc.HasMember("heightMap"))
+	{
+		const char* rawPath = doc["heightMap"].GetString();
+		//Defining variables for use
+		int error, count;
+		unsigned int resolution = 512;
+		++resolution;
+		unsigned int numVerts = resolution * resolution;
+		unsigned int numIndicies = (resolution - 1) * (resolution - 1) * 6;
+
+		//creating arrays to store data
+		std::vector<u32, fmemory::STLAllocator<u32>> heightArray;
+		heightArray.resize(numVerts);
+		//u32* heightArray = new u32[numVerts];
+		boost::container::vector<Vertex, fmemory::STLAllocator<Vertex>> terrainVertices;
+		terrainVertices.resize(numVerts);
+		boost::container::vector<u32, fmemory::STLAllocator<u32>> terrainIndices;
+		terrainIndices.resize(numIndicies);
+
+		boost::container::vector<u32, fmemory::STLAllocator<u32>> terrainOffsets;
+		terrainOffsets.push_back(0);
+
+		FILE* file;
+		error = fopen_s(&file, rawPath, "rb");
```

Comment:
fopen_s isn't available on linux, so this fails to build

Reply:
Sorry about that, I forgot to change that completely.